File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
def dfs(s, word, data):
    print(s)
    global answer
    answer += 1
    
    if s == word:
        return True
    if len(s) == 5:
        return False
    
    for a in data:
        if dfs(s+a, word, data):
            return True

def solution(word):
    global answer
    answer = 0
    
    alphabet = ['A', 'E', 'I', 'O', 'U']
    
    
    for a in alphabet:
        if dfs(a, word, alphabet):
            return answer
        

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    static int answer = 0;

    public int solution(String word) {
        String[] alphabet = {"A", "E", "I", "O", "U"};

        for (String a : alphabet) {
            if (dfs(a, word, alphabet)) {
                return answer;
            }
        }

        return 0;
    }

    public boolean dfs(String current, String target, String[] data) {
        answer++;

        if (current.equals(target)) {
            return true;
        }
        if (current.length() == 5) {
            return false;
        }

        for (String a : data) {
            if (dfs(current + a, target, data)) {
                return true;
            }
        }

        return false;
    }
}
